### PR TITLE
Add dummy libhsl_subset and libhsl_subset_64

### PR DESCRIPTION
**Pull request by amontoison**
@jfowkes @nimgould
I have two questions for you.

1) I need the following headers for the compilation:
- hsl_ma57.h
- hsl_ma87.h
- hsl_mc68.h
- hsl_precision.h
- hsl_kinds.h
- hsl_ma77.h
- hsl_ma97.h
- hsl_mi20.h
- hsl_subset_ciface.h
- hsl_ma48.h
- hsl_ma86.h
- hsl_mc64.h
- hsl_mi28.h
- hsl_subset.h
but which ones should we install?
Some of them are probably only needed for the compilation.

2- Some files (like `hsl_mi35r.f90`) require `GALAHAD_unavailable_option` from the `GALAHAD_SYMBOLS` module.
I added the `symbols.f90` file here, but can we avoid this?
I would prefer not to provide the associated Fortran module `GALAHAD_SYMBOLS` or `GALAHAD_SYMBOLS_64` with the other dummy Fortran modules.

### CODE REVIEW

**Comment by nimgould** [comment]
For 1, I suspect you only need the proper hsl_* where * is a package ones + hsl_precision.h

For 2, GALAHAD_unavailable_option is required so that GALAHAD packages can check whether they are trying to use the relevant dummy or proper hsl package. It is a nuisance, but needed.

**Comment by amontoison** [comment]
For 2, I propose to rename GALAHAD_SYMBOLS into HSL_SYMBOLS and only keep `GALAHAD_unavailable_option` here.

It insures that if you update `GALAHAD_SYMBOLS` in GALAHAD, the Fortran module generated here will not be used.
They have the same name and GALAHAD could pick this one potentially. It could be an issue if we don't backport all modifications here.

**Comment by nimgould** [comment]
This is ok, but we do need to ensure that GALAHAD_unavailable_option has the same value in your proposed new module. The _symbols on the GALAHAD side will be fixed in stone (as otherwise chaos will occur!)

**Comment by amontoison** [comment]
I think we are ready, Nick and Jari. :+1:

I created this small [`HSL_SYMBOLS`](https://github.com/ralna/libHSL/blob/hsl_subset/hsl_subset/galahad/symbols.f90) file based on your `GALAHAD_SYMBOLS`.

Once this PR is merged, I can generate a new "dummy" `HSL_jll.jl` and use it as a dependency of `GALAHAD_jll.jl` (v5.0.0). This will allow us to dynamically replace the shared library that contains the HSL symbols.

When Andrew returns in August, we can work on a new release of the official `HSL_jll.jl` that includes `libhsl_subset` and `libhsl_subset_64`. This will enable users to utilize the HSL linear solvers in the Julia interface `GALAHAD.jl` simply by installing `HSL_jll.jl`. This approach is similar to what we do with `Ipopt.jl`, and users seem quite happy with this system.

**Comment by nimgould** [comment]
That looks very good. Could you change the header so that it is
! THIS VERSION: GALAHAD 5.0 - 2024-07-09 AT 08:10 GMT.
so that it indicates that it is from GAL 5

File: hsl_subset/galahad/symbols.f90 [hunk 1]
```diff
@@ -0,0 +1,51 @@
+! THIS VERSION: GALAHAD 4.1 - 2023-02-01 AT 11:10 GMT.
```
Comment (amontoison) [inline hsl_subset/galahad/symbols.f90:1]:
```suggestion
! THIS VERSION: GALAHAD 5.0 - 2024-07-09 AT 08:10 GMT.
```

**Pull request merged**